From 1613d3fa7dbf0dc5702f9fbb5c5d59cac4d664d2 Mon Sep 17 00:00:00 2001
From: Marcel Huber <marcelhuberfoo@gmail.com>
Date: Sat, 10 Jan 2015 22:27:16 +0100
Subject: make admin utility path separate from bin path

---
 cmd/Makefile | 27 +++++++++++++++++----------
 1 file changed, 17 insertions(+), 10 deletions(-)

diff --git cmd/Makefile cmd/Makefile
index 6cfb6fb..fbf56ef 100755
--- cmd/Makefile
+++ cmd/Makefile
@@ -12,8 +12,11 @@ RM := rm
 MKDIR := mkdir -p
 
 BIN = rxadm
-DIR := /sbin
-OPT_DIR := /opt/rxadm
+BIN_DIR := /sbin
+ADM_DIR := /opt/rxadm
+ADM_NAME := pyRxAdm.py
+LOGO_DIR := /opt/rxadm
+LOGO_NAME := rxadm_logo_48x48.png
 
 .PHONY: all
 all: rxadm
@@ -22,17 +25,21 @@ all: rxadm
 .PHONY: install
 install: all
 	@echo Installing all binary files.
-	$(CP) $(BIN) $(DESTDIR)$(DIR)
-	$(CHMOD) 755 $(DESTDIR)$(DIR)/$(BIN)
-	$(MKDIR) -m 755 $(DESTDIR)$(OPT_DIR)
-	$(CP) pyRxAdm.py $(DESTDIR)$(OPT_DIR)
-	$(CP) ../misc/rxadm_logo_48x48.png $(DESTDIR)$(OPT_DIR)
+	$(MKDIR) -m 755 $(DESTDIR)$(BIN_DIR)
+	$(CP) $(BIN) $(DESTDIR)$(BIN_DIR)
+	$(CHMOD) 755 $(DESTDIR)$(BIN_DIR)/$(BIN)
+	$(MKDIR) -m 755 $(DESTDIR)$(ADM_DIR)
+	$(CP) $(ADM_NAME) $(DESTDIR)$(ADM_DIR)
+	$(CHMOD) 755 $(DESTDIR)$(ADM_DIR)/$(ADM_NAME)
+	$(MKDIR) -m 755 $(DESTDIR)$(LOGO_DIR)
+	$(CP) ../misc/$(LOGO_NAME) $(DESTDIR)$(LOGO_DIR)
 
 .PHONY: uninstall
 uninstall: 
-	@echo Uninstalling dsms files and directory path.
-	$(RM) -f $(DESTDIR)$(DIR)/$(BIN)
-	$(RM) -rf $(DESTDIR)$(OPT_DIR)
+	@echo Uninstalling files and directory path.
+	$(RM) -f $(DESTDIR)$(BIN_DIR)/$(BIN)
+	$(RM) -f $(DESTDIR)$(ADM_DIR)/$(ADM_NAME)
+	$(RM) -f $(DESTDIR)$(LOGO_DIR)/$(LOGO_NAME)
 
 msg.o: msg.c
 	$(CC) -c msg.c -o msg.o
-- 
2.2.1

