From fed893bc740d6e549b47d0bc0e6d4f795bfd6272 Mon Sep 17 00:00:00 2001
From: Marcel Huber <marcel.huber@hsr.ch>
Date: Tue, 29 Aug 2017 15:24:49 +0200
Subject: discard_zeroes_data disappears with kernel >= 4.12

---
 module/rapiddisk.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git module/rapiddisk.c module/rapiddisk.c
index a01cec6..d854936 100644
--- module/rapiddisk.c
+++ module/rapiddisk.c
@@ -508,6 +508,8 @@ rdsk_make_request(struct request_queue *q, struct bio *bio)
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,336)
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(4,8,0)
 	if (unlikely(bio_op(bio) == REQ_OP_DISCARD)) {
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(4,12,0)
+	if ((unlikely(bio_op(bio) == REQ_OP_DISCARD)) || (unlikely(bio_op(bio) == REQ_OP_WRITE_ZEROES))) {
 #else
 	if (unlikely(bio->bi_rw & REQ_DISCARD)) {
 #endif
@@ -705,7 +707,9 @@ static int attach_device(int size)
 	rdsk->rdsk_queue->nr_requests = nr_requests;
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,36)
 	rdsk->rdsk_queue->limits.discard_granularity = PAGE_SIZE;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4,12,0)
 	rdsk->rdsk_queue->limits.discard_zeroes_data = 1;
+#endif
 	rdsk->rdsk_queue->limits.max_discard_sectors = UINT_MAX;
 	queue_flag_set_unlocked(QUEUE_FLAG_DISCARD, rdsk->rdsk_queue);
 #endif
-- 
2.14.1

