From fd7a65ea2c17c2d0d423e1bf7be4111848599355 Mon Sep 17 00:00:00 2001
From: Marcel Huber <marcelhuberfoo@gmail.com>
Date: Thu, 3 Aug 2017 00:11:45 +0200
Subject: discard_zeroes_data disappears with kernel >= 4.12

---
 module/rapiddisk.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git module/rapiddisk.c module/rapiddisk.c
index a01cec6..523090a 100644
--- module/rapiddisk.c
+++ module/rapiddisk.c
@@ -705,7 +705,9 @@ static int attach_device(int size)
 	rdsk->rdsk_queue->nr_requests = nr_requests;
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,36)
 	rdsk->rdsk_queue->limits.discard_granularity = PAGE_SIZE;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4,12,0)
 	rdsk->rdsk_queue->limits.discard_zeroes_data = 1;
+#endif
 	rdsk->rdsk_queue->limits.max_discard_sectors = UINT_MAX;
 	queue_flag_set_unlocked(QUEUE_FLAG_DISCARD, rdsk->rdsk_queue);
 #endif
-- 
2.13.3

