From 8f4a9e9ee2413f09d336fb48c4f0a2ff2e617263 Mon Sep 17 00:00:00 2001
From: Marcel Huber <marcelhuberfoo@gmail.com>
Date: Tue, 14 Jun 2016 22:20:24 +0200
Subject: refactor man dir (DRY)

---
 doc/Makefile | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git doc/Makefile doc/Makefile
index 01f8c22..8b2893f 100644
--- doc/Makefile
+++ doc/Makefile
@@ -1,6 +1,7 @@
 CP := cp
 RM := rm -f
-MKDIR := mkdir -pv
+MKDIR := mkdir -pv -m 755
+MAN_DIR := /usr/share/man/man1
 
 .PHONY: all
 all: 
@@ -9,13 +10,13 @@ all:
 .PHONY: install
 install: all
 	@echo Installing rapiddisk man page.
-	$(MKDIR) $(DESTDIR)/usr/share/man/man1/
-	$(CP) rapiddisk*.1 $(DESTDIR)/usr/share/man/man1/
+	$(MKDIR) $(DESTDIR)$(MAN_DIR)
+	$(CP) rapiddisk*.1 $(DESTDIR)$(MAN_DIR)/
 
 .PHONY: uninstall
 uninstall: 
 	@echo Uninstalling rapiddisk man page.
-	$(RM) $(DESTDIR)/usr/share/man/man1/rapiddisk*.1
+	$(RM) $(DESTDIR)$(MAN_DIR)/rapiddisk*.1
 
 .PHONY: clean
 clean:
-- 
2.8.3

