From 702578bd9985f4b840920591ba4408dcdea902d2 Mon Sep 17 00:00:00 2001
From: Marcel Huber <marcelhuberfoo@gmail.com>
Date: Fri, 12 Aug 2016 11:47:10 +0200
Subject: work with kernel >=4.7.0 too

---
 module/rapiddisk.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git module/rapiddisk.c module/rapiddisk.c
index 028e229..5220758 100644
--- module/rapiddisk.c
+++ module/rapiddisk.c
@@ -667,7 +667,9 @@ static int attach_device(int size)
 	blk_queue_make_request(rdsk->rdsk_queue, rdsk_make_request);
 	blk_queue_logical_block_size(rdsk->rdsk_queue, BYTES_PER_SECTOR);
 	blk_queue_physical_block_size(rdsk->rdsk_queue, PAGE_SIZE);
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,37)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,7,0)
+	blk_queue_write_cache(rdsk->rdsk_queue, true, false);
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,37)
 	blk_queue_flush(rdsk->rdsk_queue, REQ_FLUSH);
 #else
 	blk_queue_ordered(rdsk->rdsk_queue, QUEUE_ORDERED_TAG, NULL);
-- 
2.9.2

