From ee9bcded01222124cce92e7bfabad0d4e24b953b Mon Sep 17 00:00:00 2001
From: Marcel Huber <marcelhuberfoo@gmail.com>
Date: Tue, 14 Jun 2016 22:21:03 +0200
Subject: corrected hard coded sbin path to $(DIR)

---
 src/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git src/Makefile src/Makefile
index e26a74f..70738dd 100755
--- src/Makefile
+++ src/Makefile
@@ -20,7 +20,7 @@ install: all
 	$(MKDIR) $(DESTDIR)$(DIR)
 	$(CP) $(BIN) $(DESTDIR)$(DIR)
 	$(CHMOD) 755 $(DESTDIR)$(DIR)/$(BIN)*
-	ln -s $(BIN) $(DESTDIR)/sbin/rxadm
+	ln -s $(BIN) $(DESTDIR)$(DIR)/rxadm
 
 .PHONY: uninstall
 uninstall:
-- 
2.8.3

