From 01d7a18d6c06659fe67c5bfe18ea3a275d493360 Mon Sep 17 00:00:00 2001
From: Marcel Huber <marcelhuberfoo@gmail.com>
Date: Wed, 19 Oct 2016 22:52:31 +0200
Subject: corrected hard coded sbin path to $(DIR)

---
 src/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git src/Makefile src/Makefile
index 6827b6a..10cad6e 100755
--- src/Makefile
+++ src/Makefile
@@ -30,7 +30,7 @@ install: all
 	$(MKDIR) $(DESTDIR)$(DIR)
 	$(CP) $(BIN) $(DESTDIR)$(DIR)
 	$(CHMOD) 755 $(DESTDIR)$(DIR)/$(BIN)*
-	ln -s $(BIN) $(DESTDIR)/sbin/rxadm
+	ln -s $(BIN) $(DESTDIR)$(DIR)/rxadm
 
 .PHONY: nocrypt-install
 nocrypt-install: nocrypt
@@ -38,7 +38,7 @@ nocrypt-install: nocrypt
 	$(MKDIR) $(DESTDIR)$(DIR)
 	$(CP) $(BIN) $(DESTDIR)$(DIR)
 	$(CHMOD) 755 $(DESTDIR)$(DIR)/$(BIN)*
-	ln -s $(BIN) $(DESTDIR)/sbin/rxadm
+	ln -s $(BIN) $(DESTDIR)$(DIR)/rxadm
 
 .PHONY: uninstall
 uninstall:
-- 
2.10.0

